lib/ansible/modules/network/iosxr/iosxr_lldp_global.py

```diff
@@ -0,0 +1,371 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#############################################
+#                WARNING                    #
+#############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The module file for iosxr_lldp_global
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'network'
+}
+
+DOCUMENTATION = """
+---
+module: iosxr_lldp_global
+version_added: 2.9
+short_description: Manage Global Link Layer Discovery Protocol (LLDP) settings on IOS-XR devices.
+description:
+  - This module manages Global Link Layer Discovery Protocol (LLDP) settings on IOS-XR devices.
+author: Nilashish Chakraborty (@NilashishC)
+options:
+  config:
+    description: The provided global LLDP configuration.
+    type: dict
+    suboptions:
+      holdtime:
```

Comment:
for this module, none of the params are mentioned as `required`, is that expected?

Reply:
We should probably fail, if state is not `deleted` and `config` is empty.